sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py

```diff
@@ -677,7 +700,9 @@ def _parse_event_data(self, event_data_str: str) -> Tuple[str, Any]:
             AgentStreamEvent.THREAD_RUN_CANCELLED,
             AgentStreamEvent.THREAD_RUN_EXPIRED,
         }:
-            event_data_obj = ThreadRun(**parsed_data) if isinstance(parsed_data, dict) else parsed_data
+            ParamCorrector
+            event_data_obj = ThreadRun(**ParamCorrector.filter_parameters(
```

Comment:
We probably need to check all types, like RunStep, ThreadMessage aswell

Reply:
Fixed.